packages/table/src/Row.tsx

```diff
@@ -0,0 +1,312 @@
+        if (
+          isComponentType(child, 'Row') &&
+          !hasDispatchedHasNestedRows &&
+          !hasNestedRows
+        ) {
+          dispatch({
+            type: Types.SetHasNestedRows,
+            payload: true,
+          });
+
+          hasDispatchedHasNestedRows = true;
+        }
+
+        if (
+          isComponentType(child, 'Cell') &&
+          child.props.rowSpan > 1 &&
+          !hasRowSpan &&
+          !hasDispatchedHasRowSpan
+        ) {
+          dispatch({
+            type: Types.SetHasRowSpan,
+            payload: true,
+          });
+
+          hasDispatchedHasRowSpan = true;
+        }
+      });
+    }, [children]);
+
+    // Depending on network speed, will noticeably render columns with incorrect
+    // alignment, would rather wait for proper information before rendering
+    if (!columnInfo) {
+      return null;
+    }
+
+    const chevronButton = (
+      <IconButton
+        onClick={() => setIsExpanded(curr => !curr)}
+        aria-label="chevron"
+        className={iconButtonMargin}
+      >
+        <Icon
+          aria-label="chevron"
+          glyph={isExpanded ? 'ChevronDown' : 'ChevronRight'}
+          color={uiColors.gray.dark2}
+        />
+      </IconButton>
+    );
+
+    const nestedRows = React.Children.map(children, child => {
+      if (isComponentType(child, 'Row')) {
+        return React.cloneElement(child, {
+          ref: nodeRef,
+          ['aria-expanded']: isExpanded ? 'true' : 'false',
+          isParentExpanded: isExpanded,
+          indentLevel: indentLevel + 1,
+        });
+      }
+    });
+
+    const renderedChildren = React.Children.map(children, child => {
+      if (isComponentType(child, 'CheckboxCell')) {
+        if (disabled) {
+          return React.cloneElement(child, {
+            className: disabledCell,
+            disabled,
+          });
+        } else {
+          return child;
+        }
+      } else if (isComponentType(child, 'Cell')) {
+        const { className, children } = child.props;
+
+        if (disabled) {
+          return React.cloneElement(child, {
+            className: disabledCell,
+            disabled,
+          });
+        } else {
+          if (nestedRows && nestedRows.length > 0 && !hasSeenFirstCell) {
+            hasSeenFirstCell = true;
+
+            return React.cloneElement(child, {
+              children: (
+                <>
+                  {chevronButton}
+                  <span className={truncation}>{children}</span>
+                </>
+              ),
+              className: cx(displayFlex, className),
+              key: children,
+            });
+          }
+
+          if (!children) {
+            return;
+          }
+
+          return React.cloneElement(child, {
+            children: <span className={truncation}>{children}</span>,
+          });
+        }
+      }
+    });
+
+    const shouldAltRowColor = data && data.length >= 10 && !hasNestedRows;
```

Comment:
May be my screen, but I don't see this in storybook.

Reply:
Did you look at no nested rows?

Reply:
<img width="809" alt="Screen Shot 2020-07-09 at 11 52 13 AM" src="https://user-images.githubusercontent.com/26016393/87062141-ad22ed00-c1da-11ea-83b2-819146ee7641.png">